service/controller/appcatalog/v1/resource/index/desired.go

```diff
@@ -0,0 +1,57 @@
+package index
+
+import (
+	"context"
+	"io/ioutil"
+	"k8s.io/api/core/v1"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/giantswarm/microerror"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/app-operator/pkg/label"
+	"github.com/giantswarm/app-operator/service/controller/appcatalog/v1/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	cr, err := key.ToCustomResource(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	storageURL, err := url.Parse(key.AppCatalogStorageURL(cr))
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	storageURL.Path = path.Join(storageURL.Path, "index.yaml")
+	response, err := http.Get(storageURL.String())
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	if response.StatusCode != 200 {
+		return nil, microerror.Mask(indexNotFound)
+	}
+
+	defer response.Body.Close()
+	content, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	configMap := &v1.ConfigMap{
+		ObjectMeta: v12.ObjectMeta{
+			Name:        cr.Name,
+			Namespace:   cr.Namespace,
```

Comment:
I'm not sure this is going to work! As the appCatalog CR is cluster scoped not namespace scoped. 

I think it makes sense to store it in the `giantswarm` namespace because that is where our API runs. @puja108 WDYT?

Reply:
I'm not sure either, where we should keep it?

Reply:
We should use the `giantswarm` namespace but lets add a flag for it like with the watch namespace.

`f.Service.AppCatalog.IndexNamespace` should do the job.

Reply:
We already have `f.Service.Kubernetes.Watch.Namespace` which is already using inside `app` controller, make it sense to use it here too?

Reply:
I don't think that makes sense because they have different purposes.

The watch namespace is where we check for app CRs. This namespace is where we save the index so it can be used by the API.